tests/Regressions.hs

```diff
@@ -71,6 +76,48 @@ propEqAfterDelete (Keys keys) =
 mapFromKeys :: [Int] -> HM.HashMap Int ()
 mapFromKeys keys = HM.fromList (zip keys (repeat ()))
 
+------------------------------------------------------------------------
+-- Issue #254
+
+data KC = KC Int
+  deriving (Eq, Ord, Show)
+instance Hashable KC where
+  hashWithSalt salt _ = salt
+
+issue254Lazy :: Assertion
+issue254Lazy = issue254LazyLambda 2
+
+-- Important that oldV is not hoisted out by optimisation, so use NOINLINE
+{-# NOINLINE issue254LazyLambda #-}
+issue254LazyLambda :: Int -> Assertion
+issue254LazyLambda i = do
+  _ <- return ()
+  let oldV = show i
+  weakV <- mkWeakPtr oldV Nothing
+  let mp = HML.insert (KC 1) "3" $ HML.fromList [(KC 0, "1"), (KC 1, oldV)]
+  _ <- evaluate mp
+  performGC
+  res <- deRefWeak weakV
+  _ <- evaluate mp
```

Comment:
What does this second `evaluate` do?

Reply:
Makes sure the map itself is retained after the GC. I added a comment.